ci2/ci/github.py

```diff
@@ -105,6 +109,17 @@ def to_dict(self):
         return {'repo': self.repo.to_dict(), 'name': self.name}
 
 
+class MostRecentBuild:
+    def __init__(self, source_sha, target_sha, build_state, batch):
+        self.source_sha = source_sha
+        self.target_sha = target_sha
```

Comment:
Hmm, I'm kind of against this class.  I think we should have `most_recent_batch` and `most_recent_build_state` on PR.  Ultimately the latter should go away and it should just be the batch's state itself when we're computing that.